### ISSUE TITLE
Can't use LogicEngine::saveToFile() with UTF-8-based string data on Windows 

### ISSUE THREAD

**Issue by ParadoxDaKo** [COVESA/ramses-logic#40]
**RLogic version:** 0.14.2

For file paths, we are using UTF-8-based strings to support non-ASCII data such as Umlauts or Chinese letters.

```LogicEngine::saveToFile()``` does not seem to handle UTF-8 string data, falsely decoding it in the process of saving an RLogic file on Windows systems.

Here is a GTest that elucidates the problem:

```c++
TEST(ramses_logic, saveToFile_withUmlauts) {
    rlogic::LogicEngine logicEngine;
    auto scriptContentFloat{
        R"(
function interface()
    IN.float = FLOAT
    OUT.float = FLOAT
end

function run()
    OUT.float = math.cos(IN.float)
end
)"};

    auto luaConfig = raco::ramses_base::defaultLuaConfig();
    auto* script = logicEngine.createLuaScript(scriptContentFloat, luaConfig);

    std::filesystem::path path("saveToFile_äöü.ramses");

    logicEngine.saveToFile(path.u8string());

    // use simple const char* value here
    auto status = logicEngine.loadFromFile(path.generic_string());

    ASSERT_TRUE(status);
}
```

In this test case, RLogic will save a file with the filename ```saveToFile_Ã¤Ã¶Ã¼.ramses``` but try to load a file with the filename ```saveToFile_äöü.ramses```.

**Comment by violinyanev** [comment]
That is because you are depending on platform-specific behavior ;) Straight from the docs:
![image](https://user-images.githubusercontent.com/3830505/159033071-0d2d7723-13f5-4e79-90e1-81c047f7d7dd.png)

Will document a bit better that the string_view is an ASCII string, not utf8 string. Adding non-ascii support for file saving will unnecessarily complicate the logic engine. But we can add a function to save to byte buffer, not a file, and you can use that to use your own file-string handling logic - would that work for you?

**Comment by ParadoxDaKo** [comment]
Yes, this looks like a good idea - [if there was the same functionality for Ramses](https://github.com/COVESA/ramses/issues/60) then I would be 100% happy 👍
